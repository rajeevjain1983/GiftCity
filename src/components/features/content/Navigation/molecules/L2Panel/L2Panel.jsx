// 9fbef606107a605d69c0edbcd8029e5d
import React, { useCallback, useState } from "react";
import Link from "next/link";
import withStyles from "../../../../../common/hoc/withStyle";
import { getViewportInfo } from "../../../../../../utils";
import { Row, Col, BodyCopy } from "../../../../../common/atoms";
import { keyboard } from "../../../../../../constants/keyboard.constants";
import { HideDrawerConsumer } from "../L1NavItem/L1NavItem";
import style from "./L2Panel.style";
import { CLEARANCE_ID } from "../L1NavItem/L1NavItem.config";
import L2PanelUtils from "./L2Panel.utils";

const { KEY_ENTER, KEY_SPACE } = keyboard;

const {
  MAX_ITEMS_IN_COL,
  CONDITIONAL_MAX_ITEMS_IN_COL,

  L2PropTypes,
  L2DefaultPropTypes,
} = L2PanelUtils;

const L2Panel = (props) => {
  const {
    className,
    panelData,

    l1Id,
  } = props;

  const [nestedL2Group, setNestedL2Group] = useState("");

  const handleL2GroupClick = useCallback((e) => {
    e.preventDefault();
    const groupId = e?.target?.dataset?.groupId;
    setNestedL2Group(groupId);
  });

  console.log("__ panelData", panelData);
  return (
    <HideDrawerConsumer>
      {(context) => (
        <React.Fragment>
          <div className={`${className} nav-bar-l2-panel`}>
            <div className="space-between" />
            <div className="horizontal-divider" />
            <div className="main-container">
              <Row className="content-wrapper">
                <Row
                  className="list"
                  tabIndex={-1}
                  fullBleed={{
                    small: true,
                    medium: true,
                  }}
                >
                  {panelData &&
                    panelData.map((category, categoryIndex) => {
                      const { displayName, name } = category || {};
                      return (
                        <React.Fragment>
                          <div className={`l2-nav-category`}>
                            {displayName && (
                              <div className="L2-panel-container">
                                <Link
                                  href={`c/${name}`}
                                  className="group-link"
                                  onClick={(e) => handleL2GroupClick(e)}
                                >
                                  <BodyCopy
                                    className="l2-nav-link"
                                    fontFamily="secondary"
                                    fontSize={["fs13", "fs13", "fs14"]}
                                    lineHeight="lh107"
                                    color="text.primary"
                                    data-group-id={displayName}
                                  >
                                    <span data-group-id={displayName}>
                                      {displayName}
                                    </span>
                                  </BodyCopy>
                                </Link>
                              </div>
                            )}
                          </div>
                        </React.Fragment>
                      );
                    })}
                </Row>
              </Row>
            </div>
          </div>
        </React.Fragment>
      )}
    </HideDrawerConsumer>
  );
};

L2Panel.propTypes = L2PropTypes;

L2Panel.defaultProps = L2DefaultPropTypes;

export { L2Panel as L2PanelVanilla };
export default withStyles(L2Panel, style);

{
  /* <Col colSize={{ small: 6, medium: 8, large: 12 }} ignoreGutter={{ large: true }}></Col> */
}
