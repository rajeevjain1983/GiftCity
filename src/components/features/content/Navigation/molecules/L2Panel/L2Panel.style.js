
import { css } from 'styled-components';

export default css`
  position: absolute;
  display: none;
  top: 20;
  left: 0;
  width: 100%;
  background:transparent;
  .main-container{
    background:${props => props.theme.primaryColor};
    opacity:90%;
    padding:0 ${(props) => props.theme.spacing.ELEM_SPACING.LRG} 0 30px;
    padding:20px 20px 50px 20px;
  }

.space-between{
  padding:${props=>props.theme.spacing.ELEM_SPACING.XS};
  background:transparent;
  
}
.horizontal-divider{
  background-color:${(props) => props.theme.menuHover};
   padding:4px;
   box-shadow: 0 -5px 5px -5px #F6b2f7;
}
.list{
  display: flex;
    flex-direction: column;
    flex-wrap:wrap;
    max-height: 200px;
    min-height:200px;
}


.l2-nav-link{
  display: inline-block;
  border-bottom:  3px solid transparent;
  padding-bottom:10px;
  padding:8px;
  cursor: pointer;
}
.l2-nav-link  :hover{
  
  border-bottom:  3px solid ${(props) => props.theme.bottomBorderColor};
  background-color:${(props) => props.theme.menuHover};
}

.nav-bar-l2-details{

  border:1px solid red;
}






 



  /* border:2px solid green; */
 
  /* z-index: 1200; */

  /* .L2-panel-container {
    @media ${props => props.theme.mediaQuery.large} {
      padding-right: 50px;
    }
  } */

  /* .shop-by-size-links {
    padding: 10px 14px 10px;
    ul {
      display: flex;
      flex-wrap: wrap;
      justify-content: flex-start;
      @media ${props => props.theme.mediaQuery.large} {
        flex-wrap: unset;
        justify-content: space-between;
      }
    }
  } */
  /* .shop-by-size-links li {
    cursor: pointer;
    border: 1px solid ${props => props.theme.colors.gray[800]};
    width: 40px;
    height: 40px;
    border-radius: 20px 20px;
    align-items: center;
    justify-content: center;
    display: flex;
    margin-bottom: 16px;
    margin-right: 18px;
    &:nth-child(n + 5) {
      margin-right: 0;
    }
    a {
      display: flex;
      align-items: center;
      justify-content: center;
      width: 100%;
      height: 100%;
    }
    @media ${props => props.theme.mediaQuery.large} {
      width: 38px;
      height: 38px;
    }
  } */
  /* .shop-by-size-links li:hover {
    border: solid 1px
      ${props =>
        props.theme.isGymboree
          ? props.theme.colors.orange[800]
          : props.theme.colors.blue[800]};
    background-color: ${props =>
      props.theme.isGymboree
        ? props.theme.colors.orange[100]
        : props.theme.colors.blue[100]};
  } */
 
  /* .sizes-range-background {
    min-height: 40px;
    text-align: center;
    position: relative;
    border-bottom: 1px solid ${props => props.theme.colors.gray[500]};
    margin: 0 14px;
  } */

  /* .l2-nav-link {
    display: flex;
    align-items: center;
    padding: 18px 14px 17px 28px;
    line-height: 1.15;
    &.highlighted {
      color: red;
    }
  } */

  /* .l2-nav-category-heading {
    margin: 0;
    padding: 18px 14px 17px 14px;
    background-color: ${props =>
      props.theme.isGymboree
        ? props.theme.colors.orange[50]
        : props.theme.colors.blue[50]};
  } */

  /* .s-display-none {
    display: none;
  } */

  /* .dark-overlay {
    display: none;
    background-color: green;
    position: fixed;
    top: 0;
    left: 0;
    z-index: -1;
    height: 100vh;
    opacity: 0.6;
    width: 100%;
  } */
 
  /* @media ${props => props.theme.mediaQuery.large} { */
   
    /* top: 100%;
    z-index: 2px; */
    /* border-top: 1px solid ${props => props.theme.colors.gray[300]}; */
    /* box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.25); */

    /* .s-display-none {
      display: block;
    }
    .icon-back,
    .l1-label,
    .nav-bar-item-sizes-range {
      display: none;
    }
    .l2-nav-category {
      box-sizing: border-box;
      border-right: 1px solid ${props => props.theme.colors.gray[500]};
      margin-top: 36px;
      margin-bottom: 36px;
    }
    .l2-nav-category.shop-by-size-category {
      border-right: 0;
    }
    .l2-nav-category.no-border {
      border-right: 0;
    }
    .l2-nav-category-header {
      margin-bottom: 24px;
    }
    .l2-nav-category-empty-header {
      margin-bottom: 61px;
    }
    .l2-nav-category-heading {
      padding: 0 0 15px 0;
      background: none;
    } */
    
    
  /* } */
`;
