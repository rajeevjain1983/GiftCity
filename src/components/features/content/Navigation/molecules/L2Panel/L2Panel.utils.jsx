// 9fbef606107a605d69c0edbcd8029e5d
import React from "react";
import PropTypes from "prop-types";
import { getViewportInfo } from "../../../../../../utils";
import { keyboard } from "../../../../../../constants/keyboard.constants";

const UNIDENTIFIED_GROUP = "UNIDENTIFIED_GROUP";
const MAX_ITEMS_IN_COL = 8;
const CONDITIONAL_MAX_ITEMS_IN_COL = 7;
const FOUR_COL = 4;
const TWO_COL = 2;

const NESTED_L2_CATEGORIES = ["47504", "14505"];

const renderArrowIcon = (hasSubCategories, label) => {
  return (
    hasSubCategories && <span data-group-id={label} className="icon-arrow" />
  );
};

const renderLabel = (classForRedContent, promoBadge, name) => {
  const labelToPrint = name;
  return (
    <span
      className={`nav-bar-item-label ${classForRedContent} ${
        !promoBadge ? "full-width" : ""
      }`}
    >
      {labelToPrint}
    </span>
  );
};

/**
 * This function will return whether the next column after unbxd data is shop by size or not.
 * @param {*} categoryLayout
 */
const isNextColShopBySize = (categoryLayout) => {
  if (categoryLayout) {
    const [categoryData] = categoryLayout;
    if (categoryData) {
      const { name } = categoryData;
      return name === "shopBySizeTwoColumns";
    }
  }
  return false;
};

const getColumnToDisplay = (panelCount, visibleItem) => {
  return panelCount + visibleItem > MAX_ITEMS_IN_COL ? FOUR_COL : TWO_COL;
};

const getIsNestedL2 = (l1Id, disableBabyL2Nesting) =>
  !disableBabyL2Nesting &&
  getViewportInfo() &&
  !getViewportInfo().isDesktop &&
  NESTED_L2_CATEGORIES.indexOf(l1Id) !== -1;

const getLargeColCount = (totalItemsCount) =>
  totalItemsCount > MAX_ITEMS_IN_COL ? FOUR_COL : TWO_COL;

const getColClass = (firstCol, secondCol) =>
  firstCol.length && secondCol.length ? "half-width" : "";

/* Method to close L2Drawer on keydown(ENTER and SPACE) */
const keydownHideL2Drawer = (e, hideL2Drawer) => {
  const { KEY_ENTER, KEY_SPACE } = keyboard;
  if (e.which === KEY_ENTER || e.which === KEY_SPACE) {
    hideL2Drawer(e);
  }
};

const getHideOnMobileClass = (category) =>
  category === UNIDENTIFIED_GROUP ? "s-display-none" : "";

const getNoBorderClass = (isLastPanelCol, isShopBySizeCol) =>
  isLastPanelCol && !isShopBySizeCol ? "no-border" : "";

const getTempPanelDataCount = (isLastPanelCol, tempPanelDataCount) =>
  isLastPanelCol ? 0 : tempPanelDataCount;

const getHeadingText = (nestedL2Group, name) => nestedL2Group || name;

const L2PropTypes = {
  className: PropTypes.string.isRequired,
  panelData: PropTypes.shape([]).isRequired,
  categoryLayout: PropTypes.shape([]),
  name: PropTypes.string.isRequired,
  hideL2Drawer: PropTypes.func.isRequired,
  l1Index: PropTypes.number,
  openL3Drawer: PropTypes.func.isRequired,
  hideL3Drawer: PropTypes.func.isRequired,
  closeNav: PropTypes.func.isRequired,
  l3Drawer: PropTypes.shape({}).isRequired,
  accessibilityLabels: PropTypes.shape({
    previousButton: PropTypes.string,
  }).isRequired,
  analyticsData: PropTypes.string.isRequired,
  l1Id: PropTypes.string.isRequired,
  trackNavigation: PropTypes.func.isRequired,
  isShopBySizeAbTestOff: PropTypes.bool.isRequired,
  disableBabyL2Nesting: PropTypes.bool,
};

const L2DefaultPropTypes = {
  categoryLayout: [],
  disableBabyL2Nesting: false,
  l1Index: 0,
};

export default {
  MAX_ITEMS_IN_COL,
  CONDITIONAL_MAX_ITEMS_IN_COL,
  FOUR_COL,
  TWO_COL,
  renderArrowIcon,
  renderLabel,
  isNextColShopBySize,
  getColumnToDisplay,
  getIsNestedL2,
  getLargeColCount,
  getColClass,
  keydownHideL2Drawer,
  getHideOnMobileClass,
  getNoBorderClass,
  getTempPanelDataCount,
  getHeadingText,
  L2PropTypes,
  L2DefaultPropTypes,
};
