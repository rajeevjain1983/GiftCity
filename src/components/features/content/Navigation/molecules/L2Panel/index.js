export { default } from './L2Panel';
