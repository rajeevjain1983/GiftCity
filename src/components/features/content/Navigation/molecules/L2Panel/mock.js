// 9fbef606107a605d69c0edbcd8029e5d 
const shopBySizeMockData = [
  {
    url: '/test',
    text: '4',
    title: '4',
    target: '',
  },
  {
    url: '/test',
    text: '5',
    title: '5',
    target: '',
  },
  {
    url: '/test',
    text: '6',
    title: '6',
    target: '',
  },
  {
    url: '/test',
    text: '7',
    title: '7',
    target: '',
  },
  {
    url: '/test',
    text: '8',
    title: '8',
    target: '',
  },
  {
    url: '/test',
    text: '10',
    title: '10',
    target: '',
  },
  {
    url: '/test',
    text: '12',
    title: '12',
    target: '',
  },
  {
    url: '/test',
    text: '14',
    title: '14',
    target: '',
  },
  {
    url: '/test',
    text: '16',
    title: '16',
    target: '',
  },
  {
    url: '/test',
    text: '18',
    title: '18',
    target: '',
  },
];

export default {
  shopBySizeMockData,
};
