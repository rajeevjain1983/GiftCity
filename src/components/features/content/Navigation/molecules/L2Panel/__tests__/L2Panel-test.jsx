// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import mock from '@tcp/core/src/services/abstractors/bootstrap/navigation/mock';
import Abstractor from '@tcp/core/src/services/abstractors/bootstrap/navigation';
import { shallow } from 'enzyme';
import { L2PanelVanilla as L2Panel } from '../L2Panel';

const navData = Abstractor.processData(mock.data.navigation);

const data = {
  categoryLayout: navData[0].categoryContent.mainCategory.categoryLayout,
  order: Object.keys(navData[0].subCategories),
  panelData: navData[0].subCategories,
  name: navData[0].categoryContent,
  hideL2Drawer: () => {},
  hideL3Drawer: () => {},
  accessibilityLabels: { previousButton: 'Previous' },
  className: 'nav-bar-l2',
  openL3Drawer: () => {},
};

describe('L2 Panel component', () => {
  it('renders correctly', () => {
    const L1NavItemComp = shallow(<L2Panel {...data} />);

    expect(L1NavItemComp).toMatchSnapshot();
  });

  it('has nav bar loaded', () => {
    const L1NavItemComp = shallow(<L2Panel {...data} />, { context: 'handler' }).dive();
    expect(L1NavItemComp.find('.nav-bar-l2')).toHaveLength(1);
  });
});

const dataNestedL2 = {
  categoryLayout: navData[0].categoryContent.mainCategory.categoryLayout,
  order: Object.keys(navData[0].subCategories),
  panelData: navData[0].subCategories,
  name: navData[0].categoryContent,
  hideL2Drawer: () => {},
  hideL3Drawer: () => {},
  accessibilityLabels: { previousButton: 'Previous' },
  className: 'nav-bar-l2',
  openL3Drawer: () => {},
  disableBabyL2Nesting: false,
  getViewportInfo: () => {
    return {
      isDesktop: true,
    };
  },
  l1Id: '47504',
};

describe('L2 Panel component with nested category', () => {
  it('renders correctly', () => {
    const L2NavItemComp = shallow(<L2Panel {...dataNestedL2} />);

    expect(L2NavItemComp).toMatchSnapshot();
  });

  it('has nav bar loaded', () => {
    const L1NavItemComp = shallow(<L2Panel {...dataNestedL2} />, { context: 'handler' }).dive();
    expect(L1NavItemComp.find('.nav-bar-l2')).toHaveLength(1);
  });

  it('has nav bar groups', () => {
    const L1NavItemComp = shallow(<L2Panel {...dataNestedL2} />, { context: 'handler' }).dive();
    expect(L1NavItemComp.find('.group-link')).toHaveLength(2);
  });

  it('has nav bar group click triggers list of L2', () => {
    const setNestedL2Group = jest.fn();
    const useStateSpy = jest.spyOn(React, 'useState');
    useStateSpy.mockImplementation(nestedL2Group => [nestedL2Group, setNestedL2Group]);

    const L1NavItemComp = shallow(<L2Panel {...dataNestedL2} />, { context: 'handler' }).dive();
    L1NavItemComp.find('.group-link')
      .at(1)
      .simulate('click', { preventDefault: jest.fn() });
    setTimeout(() => {
      expect(setNestedL2Group).toHaveBeenCalledWith('Clothing');
    });
  });
});
