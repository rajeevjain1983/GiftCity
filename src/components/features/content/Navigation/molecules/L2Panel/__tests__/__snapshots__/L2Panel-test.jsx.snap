// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`L2 Panel component renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`L2 Panel component with nested category renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
